src/edit.js

```diff
@@ -36,47 +53,67 @@ export default function Edit( { attributes, setAttributes } ) {
 	useEffect( () => {
 		async function getRelatedPosts() {
 			const postId = window.location.href.match( /post=([\d]+)/ )[ 1 ];
-			const response = await window.fetch( `/wp-json/wp/v2/posts?search=${ countries[ countryCode ] }&exclude=${ postId }` );
+			const response = await window.fetch(
+				`/wp-json/wp/v2/posts?filter[s]=${ countries[ countryCode ] }&exclude=${ postId }`
+			);
```

Comment:
The change introduced here is incorrect. REST API doesn't recognize a `filter` attribute and thus I'm getting all posts from the site instead of those that contain a country name in them.

<img width="233" alt="Screenshot 2022-03-31 at 15 48 32" src="https://user-images.githubusercontent.com/478735/161076328-364d90f1-e693-4d9b-9e62-5915d837439b.png">

Reply:
That being said, we were looking for replacing that entire `window.fetch` call with a more suitable approach.

In the context of the block editor, a slightly better option would be to use `@wordpress/api-fetch`. It has some built-in middlewares like the one that automatically adds nonce support.

However, related posts should be fetched using the WordPress data layer (`@wordpress/data`). Moreover, since we only need a few fields and not the entire post object, we should filter the fields we want to receive using the `_fields` property. This can be done thanks to the custom `useSelect` hook and built-in selectors (`getEntityRecords` and `getIsResolving`):

```js
	const { fetchedRelatedPosts, fetchingRelatedPosts } = useSelect(
		( select ) => {
			const getEntityRecordsArgs = [
				'postType',
				'post',
				{
					search: countries[ countryCode ],
					_fields: [ 'id', 'link', 'title', 'excerpt' ],
					exclude: postId,
				},
			];

			return {
				fetchedRelatedPosts: select( 'core' ).getEntityRecords(
					...getEntityRecordsArgs
				),
				fetchingRelatedPosts: select( 'core' ).getIsResolving(
					'getEntityRecords',
					getEntityRecordsArgs
				),
			};
		},
		[ countryCode, postId ]
	);
```

Once the related posts are retrieved, you'd only need the `setAttributes` call in this `useEffect` hook (the `useSelect` hook, just as any other custom hook, should be used outside of any React hook).